src/runtime/StateSerialization/RuntimeData.cs

```diff
@@ -250,11 +285,95 @@ private static void RestoreRuntimeDataObjects(SharedObjectsState storage)
             }
         }
 
+        static readonly string serialization_key_namepsace = "pythonnet_serialization_";
+        /// <summary>
+        /// Removes the serialization capsule from the `sys` module object.
+        /// </summary>
+        /// <remarks>
+        /// The serialization data must have been set with <code>StashSerializationData</code>
+        /// </remarks>
+        /// <param name="key">The name given to the capsule on the `sys` module object</param>
+        public static void FreeSerializationData(string key)
+        {
+            key = serialization_key_namepsace + key;
+            BorrowedReference oldCapsule = PySys_GetObject(key);
+            if (!oldCapsule.IsNull)
+            {
+                IntPtr oldData = PyCapsule_GetPointer(oldCapsule, IntPtr.Zero);
+                Marshal.FreeHGlobal(oldData);
+                PyCapsule_SetPointer(oldCapsule, IntPtr.Zero);
+                PySys_SetObject(key, null);
+            }
+        }
+
+        /// <summary>
+        /// Stores the data in the <paramref name="stream"/> argument in a Python capsule and stores
+        /// the capsule on the `sys` module object with the name <paramref name="key"/>.
+        /// </summary>
+        /// <remarks>
+        /// No checks on pre-existing names on the `sys` module object are made.
+        /// </remarks>
+        /// <param name="key">The name given to the capsule on the `sys` module object</param>
+        /// <param name="stream">A MemoryStream that contains the data to be placed in the capsule</param>
+        public static void StashSerializationData(string key, MemoryStream stream)
+        {
+            var data = stream.GetBuffer();
```

Comment:
This might have extra data in the end

Reply:
What do you mean by extra data? The stream container is longer than the data in it? The code is using `Length` not `Capacity`, so it should not be an issue, no?